src/server.c

```diff
@@ -1207,6 +1210,10 @@ static qd_failover_item_t *qd_connector_get_conn_info_lh(qd_connector_t *ct) TA_
 /* Timer callback to try/retry connection open, connector->lock held */
 static void try_open_lh(qd_connector_t *connector, qd_connection_t *connection) TA_REQ(connector->lock)
 {
+    // If the connector has been deleted, do not try to open new connection.
```

Comment:
This shouldn't ever happen, right?  The caller to try_open_lh explicitly checks that the state is valid before making this call.  You could replace this check with an assert just to prevent us from breaking that calling requirement.